### DISCUSSION TITLE
Bulk processing, resizing and renaming

### DISCUSSION THREAD

**Discussion by autografAU** [ImageMagick/ImageMagick#6505] [Help]
### ImageMagick version

7

### Operating system, version and so on

Windows

### Description

I haven't used this program yet...but just wondering if it can do something like: take every image in a folder...resize them to particular dimensions...then saves them in the same folder but with a different name. 

So for example, if the original file was an image 1920px X 1080px called photo.jpg, the script would resize it to 350px height and the scaled width and then save it as photo_th.jpg with a particular quality.

Is that something it could handle? Excuse my ignorance.

**Comment by snibgo** [comment]
Yes. For example, this processes a single file:
```
magick in.png -resize x350 -set filename:MYFILE %t_th.jpg %[filename:MYFILE]
```
It resizes to make the height 350 pixels, and the appropriate width so the aspect ratio doesn't change . The output filename is the same as the input but with `_th`, and a `jpg` extension.

You can put that in a shell `for` loop to process all the files in a directory. Beware: if you do this more than once, you will get thumbnails of thumbnails. Then thumbnails of thumbnails of thumbnails ... You could avoid this by inserting a test so you don't process files like `*_th.*`. Or you could write the thumbnails to a different directory.
